packages/dds/tree/src/simple-tree/schemaFactory.ts

```diff
@@ -300,17 +301,18 @@ export class SchemaFactory<
 			public constructor(input: FlexTreeNode | unknown) {
 				super();
 				// Currently this just does validation. All other logic is in the subclass.
-				if (isFlexTreeNode(input)) {
-					assert(
-						getClassSchema(input.schema) === this.constructor,
-						0x83b /* building node with wrong schema */,
+				if (isFlexTreeNode(input) && getClassSchema(input.schema) !== this.constructor) {
+					throw new TypeError(
+						`The provided input is incompatible with "${schema.identifier}" schema.`,
```

Comment:
It's a pet peeve of mine to surround dynamic string content in error messages with single quotes, to avoid weird edge cases where a string could be empty and the error message the user sees makes no sense or acquires a different meaning, vs a clear signal of "we got a blank". Not sure that can happen in this case (`schema.identifier` being empty), just leaving it for consideration.

Reply:
I'm actually not sure I'm following the suggestion. Your argument seems to be in favor of surrounding the value in quotes (which I did).

Reply:
Oh, there's the double quotes.... sorry, yes, you already did what I was saying. I missed that the main string is defined with backticks and parsed the double quotes as being the limit of the first part of the full string. My bad :)

Reply:
Haha, that's okay. For a bit I thought you were saying you didn't like the quotes, but as I kept reading it kept seeming like we had the same reasons for preferring them 🤣